### ISSUE TITLE
Allow unchanged deploys

### ISSUE THREAD

**Issue by nahtnam** [superfly/flyctl#843]
*Please only report specific issues with `flyctl` behavior*. Anything like a support request for your application should go to https://community.fly.io. More people watch that space and can help you faster!

**Describe the bug**

You can see more context here: https://community.fly.io/t/error-app-is-unchanged-skipping-release/4154

The fly command used to not throw any error if you were deploying the same code. Now it does, which is fine but get annoying when working with CI. Would love to have a flag such as `--allow-unchanged` or `--ignore-unchanged` or something like that, where it does not throw an error allowing the CI to pass

** Command output: **

```text
Error App is unchanged, skipping release
```

**Comment by charsleysa** [comment]
Related https://github.com/superfly/flyctl-actions/issues/17

**Comment by chris-teamout** [comment]
Hi all,
I see a few people have 👍  @charsleysa's ticket. Just wanted to check if there is any update/discussion ongoing about a flag. We are also facing some issues in our Github actions.

Any suggestion on how to resolve? Happy to contribute if needed!
Thanks for your guidance 🙌

**Comment by jsierles** [comment]
Right now this isn't being worked on, but we welcome PRs!

**Comment by chris-teamout** [comment]
@jsierles,

I looked at the flyctl code and I would need a few pointers.

```golang
	cmdfmt.PrintBegin(cmdCtx.Out, "Creating release")

	input := api.DeployImageInput{
		AppID: cmdCtx.AppName,
		Image: img.Tag,
	}
	if val := cmdCtx.Config.GetString("strategy"); val != "" {
		input.Strategy = api.StringPointer(strings.ToUpper(val))
	}
	if cmdCtx.AppConfig != nil && len(cmdCtx.AppConfig.Definition) > 0 {
		input.Definition = api.DefinitionPtr(cmdCtx.AppConfig.Definition)
	}

	release, releaseCommand, err := cmdCtx.Client.API().DeployImage(ctx, input)
	if err != nil {
		return err
	}

	fmt.Fprintf(cmdCtx.Out, "Release v%d created\n", release.Version)
	if releaseCommand != nil {
		fmt.Fprintf(cmdCtx.Out, "Release command detected: this new release will not be available until the command succeeds.\n")
	}
```
> source: deploy.go (both in `/cmd` and `/internal/command`)

I think it's coming from the `DeployImage()` function which sends a graphql request to fly and it's the backend side which responds `Error App is unchanged, skipping release`

If that's correct, can you point to the repository where to look at the backend? I can suggest to add the cobra/viper flag `--ignore-unchanged` and pass the parameter to the backend using the `DeployImageInput` structure.

Let me know what you think!

**Comment by jsierles** [comment]
Hey, sorry for the delay, this one slipped. It turns out we have a branch for this internally and may be able to get it out soon. @michaeldwan

**Comment by jsierles** [comment]
This was completed in #1055 and released in [0.0.347](https://github.com/superfly/flyctl/releases/tag/v0.0.347).

**Comment by dentarg** [comment]
Sounds like this issue can be closed?
